### PR TITLE
Fix tag overwriting in loop

### LINKED ISSUE #67: BUG: Fix tag overwriting in loop
In `app.py` lines 130-140, the function `process_tags` overwrites an evaluated variable in the loop.

Redefinition of a loop variable inside the loop's body causes its value to differ from the original loop iteration for the remainder of the block, in a way that will likely cause bugs.

In Python, unlike many other languages, for loops and with statements don't define their own scopes. Therefore, a nested loop that uses the same target variable name as an outer loop will reuse the same actual variable, and the value from the last iteration will "leak out" into the remainder of the enclosing loop.

While this mistake is easy to spot in small examples, it can be hidden in larger blocks of code, where the definition and redefinition of the variable may not be visible at the same time.

```python
def process_tags(tags, topic):
    processed = []
    for tag in tags:
        # Clean tag
       >>> tag = tag.lower().strip()
        >>> tag = re.sub(r'[^\w\s-]', '', tag)  # Remove special chars except - and space

        # Skip if too short or doesn't contain topic
        if len(tag) < 3 or topic.lower() not in tag:
            continue

        processed.append(tag)
    return processed
```

### PR DESCRIPTION
**Pull request by dakotatokad**
- Refactor tag processing into new module
- Add doctoring
- Fix variable redefinition inside of the loop
- Add associated unit tests

Closes #67

### CODE REVIEW

**Review by dakotatokad — COMMENTED** [review]
Personal fix-em-up notes

File: src/tag_processing.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    Args:
+        tags (list[str]): A list of possible tags to filter.
+        topic (str): The topic to filter tags by.
+
+    Returns:
+        list[str]: A list of cleaned and relevant tags relevant to the topic.
+    """
+    processed_tags = []
+    minimum_tag_length = 3
```
[the remark this answers is not in the document]
Comment (dakotatokad) [reply src/tag_processing.py:20]: Change to variable with default

**Comment by imDarshanGK** [comment]
Great work @dakotatokad, on refactoring the tag processing logic into its own module and adding clear docstrings and type hints. The new unit tests are helpful, and the code is much more modular now!

Suggestions:

1) After replacing special characters in tags, consider normalizing whitespace (e.g., using ' '.join(clean_tag.split())) to avoid multiple spaces in tags.
2) For the minimum tag length, you might want to use >= instead of > if you want to include tags that are exactly the minimum length.
3) The test cases cover a good range consider adding an edge case where the topic is empty or whitespace.

Waiting for all required checks to complete and review comments to be resolved before merging. Will proceed once everything is green!

Thanks for the clear commit history and thorough testing!

**Comment by dakotatokad** [comment]
@imDarshanGK good to go

**Review by imDarshanGK — APPROVED** [review]
Looks great @dakotatokad, thanks for your contribution!
